client/src/pages/SplashScreen/SplashScreen.tsx

```diff
@@ -0,0 +1,27 @@
+
+import { useState, useEffect } from "react";
+import { ScreenOne, ScreenTwo } from "../../components";
+import { useNavigate } from 'react-router-dom';
+
+export const SplashScreen = () => {
+    const [next_screen, setNext_screen] = useState("first_screen")
+    const navigate = useNavigate();
+    const handleNavigation = () => {
+        navigate('/welcome');
+    }
+    useEffect(() => {
+        if (next_screen == "welcome") {
+            handleNavigation()
+        }
+    }, [next_screen])

```

Comment:
I am not sure I understand the need for this code.
Why can we use `useNavigate` directly inside ScreenOne && ScreenTwo respectively?

We just need to import it into those components and navigate to the different screens

Reply:
I wanted the useEffect() to be just like a middleware for the splash screen, so when the splash screen is done it moves to the next page, in screenTwo, when the animation finishes it changes the value of next_screen to welcome, so once the value is set to welcome it moves to the next page, so we can make navigate dynamic like this: navigate(dynamic_variable)

Reply:
why not just navigate to welcome directly from the SplashScreenTwo component?

```js
export const ScreenTwo = ({ setNext_screen }: Props) => {
const navigate = useNavigate();
    return (
        <div id={styles.wrapper}>

            <img src={logo} />
            <div className={styles.progress_bar_background}>

                <motion.div
                    // transition={increaseWidth}
                    initial={{ width: '0%' }}
                    animate={{ width: '100%' }}
                    transition={{ duration: 10 }}
                    onAnimationComplete={() => navigate('/welcome')}
                    className={styles.progress_bar} ></motion.div>
            </div>
        </div>

    )
}

```
The useNavigate should work because the parent component is wrapped inside a Router.
You can then decide to conditionally load the different screen based on a change like you are doing.

Also, for the `useState` can you add a type to what kind of data it receives, currently, it receives any kind of string. So the code might easily break because I can add setNextScreen('hello'). This will break the code.

Create a Types folder and Add something like this 
`export type SplashScreenState = 'first_screen' | 'progress_bar' ;`

Then for the useState inside the SplashScreen Page, do something like this:
`const [nextScreen, setNextScreen] = useState<SplashScreenState>("first_screen")`

Your prop type for setNextScreen in screen one and two will no longer be `React.Dispatch<React.SetStateAction<string>>` but `React.Dispatch<React.SetStateAction<SplashScreenState>>`

Reply:
ok